music-journal/components/ListMemories.js

```diff
@@ -36,10 +31,10 @@ const ListMemories = ({ firebaseToken }) => {
               </tr>
             </thead>
             <tbody id="memoriestablebody">
-              {memories.length > 0 && memories.map(memory => 
-                <tr key={memory.song}>
-                  <td>{memory.song}</td>
-                  <td>{memory.text}</td>
+              {memories.length > 0 && memories.map(({song, text}) => 
```

Comment:
So this will essentially map every song and text value within memories to a bit of html which is the table row? 🤯

Reply:
Yep, pure functions are a big deal in React like map, reduce, filter as they return a new object rather than then original one so it helps keep the state loop working properly.